### DISCUSSION TITLE
.NET MAUI - MS SQL Server Connection workding with Windows, not working with Android.

### DISCUSSION THREAD

**Discussion by anwarulmustafa** [dotnet/maui#10049] [Q&A]
Hi, I am trying to connect my .net maui app to ms sql server. Here is my connection string. 

"Data Source=192.168.100.12;Initial Catalog=LogicSSS;User ID=myuser;Password=P45678$;Encrypt=False;TrustServerCertificate=True;";

I am connected with windows but when i run my application in android. it gives me an error. 

A connection was successfully established with the server, but then an error occurred during the pre-login handshake. (provider: TCP Provider, error: 35 - An internal exception was caught)

I need guidence.

Thanks
Awar

**Comment by jefffeely** [comment]
I have run into this problem repeatably so I am sad no one has given an answer.  In my research the option "TrustServerCertificate=True;"" (which you use) is supposed to fix that error.  

Another thing I came up with is that Android 7 has a critical bug which prevents the certificate from being trusted no matter what you do.  The fix for that is to just not use Android 7.  

There is a new version of the mssql client that came out recently.  Maybe that will have fixed it?

**Comment by dmccolloughOneGas** [comment]
You do understand that it's a really, really, really poor design decision to tightly couple your .NET Maui App directly to SQL Server.

A better approach would be to host a API somewhere, on-premise, in Azure, in AWS and have your app talk to the API. The API would then communicate directly to the SQL Server.

**Reply by yasseryoussef** [reply]
there are some applications have no need to use api if you are working on a caffe application which run in local network why you well need to use api

**Comment by Xyncgas** [comment]
Have you tried postgresql, if it doesn't work try restarting your computer
